### PR TITLE
boards/arduino-mkrzero: add initial support

### PR DESCRIPTION
**Pull request by aabadie**
And factorize code with arduino-mkr1000 since these boards are very similar.
The Arduino MKRZERO uses the second SPI to connect a SD card device, I configured it to use the one available in RIOT. This is not yet tested though.

I also noticed that there's a third one in the family: [Arduino MKRFOX1200](https://www.arduino.cc/en/Main.ArduinoBoardMKRFox1200) that offers a SigFox compatible antenna.

### CODE REVIEW

**Comment by aabadie** [comment]
And this PR is based on #6666

**Comment by aabadie** [comment]
since #6666 has been merged, I rebased this one. Tested mkr1000, still works.

File: boards/arduino-mkrzero/include/board.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    SD Card device configuration
+ * @{
+ */
+#define SDCARD_SPI_PARAM_SPI         (SPI_DEV(1))
```
Comment (aabadie) [inline boards/arduino-mkrzero/include/board.h]: This is the part, I'm not totally sure of. @haukepetersen or @vincent-d any comments?

Comment (haukepetersen) [inline boards/arduino-mkrzero/include/board.h]: Depends on what the question is... Does the `arduino-mkrzero` come with an on-board SD-card slot? If yes, then this looks valid to me (as long as the SD-card slot is connected to the pins mapped  for SPI_DEV(1)...).

**Comment by haukepetersen** [comment]
before merging this PR, I would however like to see a better solution for the `bossa` tool, as having the 4th board shipped with the same binary tool is ugly and bad -> so I would like to have a better solution before we go on. Sounds fair?

**Comment by aabadie** [comment]
> Sounds fair?

Yes, I'm interested in looking into this except if you feel very motivated.

File: boards/arduino-mkrzero/include/board.h (hunk 1 above)
Comment (aabadie) [inline boards/arduino-mkrzero/include/board.h]: It should be ok, regarding the [specs](https://www.arduino.cc/en/uploads/Main/ArduinoMKRZero-schematic.pdf). I just don't know what CD1 is on the SD card reader.

Comment (gebart) [inline boards/arduino-mkrzero/include/board.h]: Could be "card detect"?

Comment (gebart) [inline boards/arduino-mkrzero/include/board.h]: Could be "card detect"?

**Comment by haukepetersen** [comment]
done: #7068

File: boards/arduino-mkr-common/include/periph_conf_common.h [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        .mosi_pad = SPI_PAD_MOSI_0_SCK_1
+    },
+    {
+        .dev      = &SERCOM2->SPI,
+        .miso_pin = GPIO_PIN(PA, 15),
+        .mosi_pin = GPIO_PIN(PA, 12),
+        .clk_pin  = GPIO_PIN(PA, 13),
+        .miso_mux = GPIO_MUX_D,
+        .mosi_mux = GPIO_MUX_D,
+        .clk_mux  = GPIO_MUX_D,
```
Comment (kbumsik) [inline boards/arduino-mkr-common/include/periph_conf_common.h]: I noticed my silly mistake while testing WINC1500 WiFi module. 🤕 Could you please change the MUX pads to:

```C
.miso_mux = GPIO_MUX_C,
.mosi_mux = GPIO_MUX_C,
.clk_mux  = GPIO_MUX_C,
```
Thank you.

Comment (aabadie) [inline boards/arduino-mkr-common/include/periph_conf_common.h]: no problem, directly squashed and pushed

File: boards/arduino-mkr-common/include/periph_conf_common.h [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        .clk_pin  = GPIO_PIN(PA, 17),   /* ARDUINO_PIN_9, SERCOM1-SCK */
+        .miso_mux = GPIO_MUX_C,
+        .mosi_mux = GPIO_MUX_C,
+        .clk_mux  = GPIO_MUX_C,
+        .miso_pad = SPI_PAD_MISO_3,
+        .mosi_pad = SPI_PAD_MOSI_0_SCK_1
+    },
+    {
+        .dev      = &SERCOM2->SPI,
+        .miso_pin = GPIO_PIN(PA, 15),
```
Comment (aabadie) [inline boards/arduino-mkr-common/include/periph_conf_common.h]: This pin definition for the second SPI is different from mkr1000. I'll have to provide the SPI configuration for each boards.

**Comment by haukepetersen** [comment]
now that #7068 is merged, would you mind to rebase and make use of it?

**Comment by aabadie** [comment]
>  would you mind to rebase and make use of it?

I'm on it

File: boards/arduino-mkrzero/include/board.h (hunk 1 above)
Comment (aabadie) [inline boards/arduino-mkrzero/include/board.h]: Thanks, googling points toward this direction. I added a define for this pin below in the file.

**Comment by aabadie** [comment]
@haukepetersen rebased and directly squashed

**Comment by haukepetersen** [comment]
Looks good to me, though I have no board to test this... Anyone else?

**Comment by aabadie** [comment]
anyone for testing this ?

**Comment by aabadie** [comment]
@astralien3000, will you have time tomorrow for testing this one ? (I can lend you mine)

**Comment by astralien3000** [comment]
First, bossa does not compile on MacOS

**Comment by aabadie** [comment]
>  bossa does not compile on MacOS

I think @kYc0o opened a PR about that some days ago.

**Comment by kYc0o** [comment]
You can ACK my PR #7191 😃

**Comment by aabadie** [comment]
@astralien3000, since the mac OS problem with is unrelated, can you try with a Linux VM ? At least, we can move forward here.

**Comment by astralien3000** [comment]
Sorry, but I can't flash it a all now (neither Mac nor Linux VM), I can't put it in flashing mode...

**Comment by astralien3000** [comment]
I tested `default` and `xtimer_usleep`, OK.
But `periph_timer` and `periph_pwm` don't seem to work (LED is oscillating, though)

**Comment by aabadie** [comment]
are you on the most recent commit ?

**Comment by astralien3000** [comment]
hmm... I don't know, but since you did not add commits after my first post, I think I was.
I will check this on Monday (I don't have the computer I used for this at home).

**Comment by aabadie** [comment]
Maybe there's a problem with PWM. Will check again when time permits. Thanks for testing !

**Comment by astralien3000** [comment]
I have checked, I was on the most recent commit. And I also needed to rebase to be able to flash the board (bossa problems with MacOS...).

**Comment by aabadie** [comment]
@astralien3000, I pushed a fox regarding the PWM: pin ~2 and ~3 are working but I couldn't find the correct conf for pin ~4 and ~5. So there's only 2 available PWM pins in the default configuration.

Regarding the periph_timer test, I didn't test but it should work as I didn't change it and is working (normally) with mkr1000.

**Comment by aabadie** [comment]
@astralien3000, if you have some time can you give another try to this one ?

**Comment by astralien3000** [comment]
- `periph_pwm` OK (for ~2 and ~3)
 - `periph_timer` still blocking

**Comment by aabadie** [comment]
@astralien3000, thanks for testing again.

Regarding the timer test issue, I fixed it: by default mkrzero use the `sdcard_spi` module which itself uses xtimer and for some reason this messes up this test (TIMER_0 being initialized with different values apparently).
The fix is simply to remove the dependency to `sdcard_spi` which makes sense since this is up to the user to include it or not.

In the mean time, I also fixed the second spi periph configuration which was wrong.

Directly squashed and rebased.

Can you test again ?

**Comment by astralien3000** [comment]
Yes ! Now it works !

**Comment by aabadie** [comment]
@haukepetersen, this one is ready now (and tested). Would be great if you could approve it (and merge ;)) !

**Comment by aabadie** [comment]
@haukepetersen ?

File: boards/arduino-mkr-common/include/arduino_pinmap.h [hunk 4]
```diff
@@ -56,7 +56,7 @@ extern "C" {
 #define ARDUINO_PIN_A4          GPIO_PIN(PA, 5)  /* AIN5 */
 #define ARDUINO_PIN_A5          GPIO_PIN(PA, 6)  /* AIN6 */
 #define ARDUINO_PIN_A6          GPIO_PIN(PA, 7)  /* AIN7 */
-/** @} */
+/** @ */
```
Comment (haukepetersen) [inline boards/arduino-mkr-common/include/arduino_pinmap.h]: any reason for removing the `}`?

File: boards/arduino-mkr-common/include/arduino_pinmap.h [hunk 5]
```diff
@@ -29,7 +29,7 @@ extern "C" {
 #endif
 
 /**
- * @name    Mapping of MCU pins to Arduino pins
+ * @name   Mapping of MCU pins to Arduino pins
```
Comment (haukepetersen) [inline boards/arduino-mkr-common/include/arduino_pinmap.h]: Indention off by one space.

File: boards/arduino-mkr-common/include/board_common.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+extern "C" {
+#endif
+
+/**
+ * @name    xtimer configuration
+ * @{
+ */
+#define XTIMER              TIMER_0
+#define XTIMER_CHAN         (0)
+/** @} */
```
Comment (haukepetersen) [inline boards/arduino-mkr-common/include/board_common.h]: please remove this block - no need to re-define the default configuration

File: boards/arduino-mkr1000/include/board.h [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2016 Inria
+ * Copyright (C) 2017 Inria
```
Comment (haukepetersen) [inline boards/arduino-mkr1000/include/board.h]: Actually, I think convention is to use `2016-2017` when editing. Applies to many files of this PR

**Comment by aabadie** [comment]
@haukepetersen, comments addressed.

**Comment by haukepetersen** [comment]
all green -> go

**Pull request merged**